File: PasswordList/src/entity/PasswordItem.java
package entity;

import java.text.SimpleDateFormat;
import java.util.Date;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

/**
 * パスワード情報を記録
 */
public class PasswordItem {

	// 最長の文字数
	static public final int MAXLENGTH_Name = 128;
	static public final int MAXLENGTH_UserID = 128;
	static public final int MAXLENGTH_Password = 128;
	static public final int MAXLENGTH_Comment = 512;
	static public final int MAXLENGTH_MailAddress = 128;

	/**
	 * 識別ID
	 */
	private int ItemID = 0;

	/**
	 * 名前
	 */
	private String Name = null;

	/**
	 * 登録時間(秒まで)
	 */
	private Date RegisterDate = null;

	/**
	 * 更新時間(秒まで)
	 */
	private Date UpdateDate = null;

	/**
	 * ユーザーID
	 */
	private String UserID = null;

	/**
	 * パスワード
	 */
	private String Password = null;

	/**
	 * コメント
	 */
	private String Comment = null;

	/**
	 * メールアドレス
	 */
	private String MailAddress = null;

	// ----------------------------------------------------------------------

	// デフォルトコンストラクタ
	public PasswordItem() {
	}

	/**
	 * オブジェクトの複製を行う【コピーコンストラクタ】
	 *
	 * @param item コピー元オブジェクト
	 */
	public PasswordItem(PasswordItem item) {
		ItemID = item.ItemID;
		Name = item.Name;
		if (item.RegisterDate != null) RegisterDate = new Date(item.RegisterDate.getTime());
		if (item.UpdateDate != null) UpdateDate = new Date(item.UpdateDate.getTime());
		UserID = item.UserID;
		Password = item.Password;
		Comment = item.Comment;
		MailAddress = item.MailAddress;
	}

	// ----------------------------------------------------------------------

	/**
	 * 識別IDを取得する
	 *
	 * @return 識別ID(0は無効値)
	 */
	public int getItemID() {
		return ItemID;
	}

	/**
	 * 識別IDを設定する
	 *
	 * @param itemID 設定する識別ID(0以下は無効値)
	 */
	public void setItemID(int itemID) {
		if (itemID < 0) itemID = 0;
		ItemID = itemID;
	}

	/**
	 * 名前を取得する
	 *
	 * @return 名前(nullは無効値)
	 */
	public String getName() {
		return Name;
	}

	/**
	 * 名前を設定する
	 *
	 * @param 名前(nullは無効値)
	 */
	public void setName(String name) {
		Name = name;
	}

	/**
	 * 登録日を取得する(Json化では使用されない)
	 *
	 * @return 登録日(nullは無効値)
	 */
	@JsonIgnore
	public Date getRegisterDate() {
		return RegisterDate;
	}

	/**
	 * 登録日を取得する(Json化では使用されない)
	 *
	 * @return 登録日(nullは無効値)
	 */
	@JsonProperty("RegisterDateTime")
	public String getRegisterDateStr() {
		if (getRegisterDate() == null) return null;
		// SimpleDateFormatクラスでフォーマットパターンを設定する
		SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd HHmmss");
		return sdf.format(getRegisterDate());
	}

	/**
	 * 登録日を設定する
	 *
	 * @param 登録日(nullは無効値)
	 */
	public void setRegisterDate(Date registerDate) {
		RegisterDate = new Date(registerDate.getTime());
	}

	/**
	 * 更新日を取得する
	 *
	 * @return 更新日(nullは無効値)
	 */
	@JsonIgnore
	public Date getUpdateDate() {
		return UpdateDate;
	}

	/**
	 * 更新日を取得する
	 *
	 * @return 更新日(nullは無効値)
	 */
	@JsonProperty("UpdateDateTime")
	public String getUpdateDateStr() {
		if (getRegisterDate() == null) return null;
		// SimpleDateFormatクラスでフォーマットパターンを設定する
		SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd HHmmss");
		return sdf.format(getUpdateDate());
	}

	/**
	 * 更新日を設定する
	 *
	 * @param 更新日(nullは無効値)
	 */
	public void setUpdateDate(Date updateDate) {
		UpdateDate = new Date(updateDate.getTime());
	}

	/**
	 * ユーザーIDを取得する
	 *
	 * @return ユーザーID(nullは無効値)
	 */
	public String getUserID() {
		return UserID;
	}

	/**
	 * ユーザーIDを設定する
	 *
	 * @param ユーザーID(nullは無効値)
	 */
	public void setUserID(String userID) {
		UserID = userID;
	}

	/**
	 * パスワードを取得する
	 *
	 * @return パスワード(nullは無効値)
	 */
	public String getPassword() {
		return Password;
	}

	/**
	 * パスワードを設定する
	 *
	 * @param パスワード(nullは無効値)
	 */
	public void setPassword(String password) {
		Password = password;
	}

	/**
	 * コメントを取得する
	 *
	 * @return コメント(nullは無効値)
	 */
	public String getComment() {
		return Comment;
	}

	/**
	 * コメントを設定する
	 *
	 * @param コメント(nullは無効値)
	 */
	public void setComment(String comment) {
		Comment = comment;
	}

	/**
	 * メールアドレスを取得する
	 *
	 * @return メールアドレス(nullは無効値)
	 */
	public String getMailAddress() {
		return MailAddress;
	}

	/**
	 * メールアドレスを設定する
	 *
	 * @param getMailAddress
	 */
	public void setMailAddress(String mailAddress) {
		MailAddress = mailAddress;
	}
}

File: PasswordListAndroid/app/src/main/java/android/ochawanz/passwordlistandroid/jsonobject/PasswordItem.kt
package android.ochawanz.passwordlistandroid.jsonobject

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

/**
 * パスワードアイテム
 */
@JsonIgnoreProperties(ignoreUnknown = true)
class PasswordItem() {
	companion object {
		// 最長の文字数
		val MAXLENGTH_Name = 128
		val MAXLENGTH_UserID = 128
		val MAXLENGTH_Password = 128
		val MAXLENGTH_Comment = 512
		val MAXLENGTH_MailAddress = 128
	}

	/**
	 * コピーコンストラクタ
	 * @param passwordItem コピー元インスタンス
	 */
	constructor(passwordItem: PasswordItem) : this() {
		ItemID = passwordItem.ItemID
		Name = passwordItem.Name
		RegisterDate = passwordItem.RegisterDate
		UpdateDate = passwordItem.UpdateDate
		UserID = passwordItem.UserID
		Comment = passwordItem.Comment
		MailAddress = passwordItem.MailAddress
		Password = passwordItem.Password
	}

	/**
	 * 識別ID(0は無効値)
	 */
	@field:JsonProperty("ItemID")
	var ItemID: Int = 0
		set(value) {
			field = if (value < 0) 0 else value
		}

	/**
	 * 名前
	 */
	@JsonProperty("Name")
	var Name: String? = null

	/**
	 * 登録日
	 */
	@JsonIgnore
	var RegisterDate: Date? = null

	/**
	 * 更新日
	 */
	@JsonIgnore
	var UpdateDate: Date? = null

	/**
	 * ユーザーID
	 */
	@JsonProperty("UserID")
	var UserID: String? = null

	/**
	 * コメント
	 */
	@JsonProperty("Comment")
	var Comment: String? = null

	/**
	 * メールアドレス
	 */
	@JsonProperty("MailAddress")
	var MailAddress: String? = null

	/**
	 * パスワード
	 */
	@JsonProperty("Password")
	var Password: String? = null

	/**
	 * 登録日の文字列版(yyyyMMdd HHmmss)
	 */
	@set:JsonProperty("RegisterDateTime")
	@get:JsonProperty("RegisterDateTime")
	var RegisterDateWithString: String?
		get() = if (RegisterDate == null) null else SimpleDateFormat("yyyyMMdd HHmmss").format(RegisterDate)
		set(value) {
			if (value == null) {
				RegisterDate = null
				return
			}
			RegisterDate = try {
				SimpleDateFormat("yyyyMMdd HHmmss").parse(value)
			} catch (e: ParseException) {
				null
			}
		}

	/**
	 * 更新日の文字列版(yyyyMMdd HHmmss)
	 */
	@set:JsonProperty("UpdateDateTime")
	@get:JsonProperty("UpdateDateTime")
	var UpdateDateWithString: String?
		get() = if (UpdateDate == null) null else SimpleDateFormat("yyyyMMdd HHmmss").format(UpdateDate)
		set(value) {
			if (value == null) {
				UpdateDate = null
				return
			}
			UpdateDate = try {
				SimpleDateFormat("yyyyMMdd HHmmss").parse(value)
			} catch (e: ParseException) {
				null
			}
		}


	override fun toString() = Name + "(#" + java.text.DecimalFormat("0000").format(ItemID) + ")"
}